hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -187,6 +188,14 @@ public void initialize(URI name, Configuration conf) throws IOException {
       awsConf.setSignerOverride(signerOverride);
     }
 
+    s3IsRequesterPays = conf.getBoolean(ALLOW_REQUESTER_PAYS,
+        DEFAULT_ALLOW_REQUESTER_PAYS);
+
+    if (s3IsRequesterPays == true) {
+      LOG.info("fs.s3a.requester-pays.enabled has been set to true. You will be charged for any requests made to Requester Pays buckets.");
+      LOG.info("For more information on Requester Pays buckets visit: http://docs.aws.amazon.com/AmazonS3/latest/dev/RequesterPaysBuckets.html");
+    }
```

Comment:
How is the below code logic achieved here?

`awsConf.addHeader("x-amz-request-payer", "requester");`

Reply:
Previous version of AWS Java SDK(1.10.6) achieves this through the flag in the 
GetObjectRequest request object and internally it uses this flag.
In AmazonS3Client .getObject() (again this is amazon sdk class)
   if (getObjectRequest.isRequesterPays()) {
      request.addHeader("x-amz-request-payer", "requester");
    }

Reply:
Cool, thanks for the explanation. lgtm.